beacon-chain/state/state-native/state_trie.go

```diff
@@ -0,0 +1,1060 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "could not compute historical roots merkleization")
+	}
+	fieldRoots[fieldRootIx] = historicalRootsRt[:]
+	fieldRootIx++
+
+	// Eth1Data data structure root.
+	eth1HashTreeRoot, err := stateutil.Eth1Root(hasher, state.eth1Data)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not compute eth1data merkleization")
+	}
+	fieldRoots[fieldRootIx] = eth1HashTreeRoot[:]
+	fieldRootIx++
+
+	// Eth1DataVotes slice root.
+	eth1VotesRoot, err := stateutil.Eth1DataVotesRoot(state.eth1DataVotes)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not compute eth1data votes merkleization")
+	}
+	fieldRoots[fieldRootIx] = eth1VotesRoot[:]
+	fieldRootIx++
+
+	// Eth1DepositIndex root.
+	eth1DepositIndexBuf := make([]byte, 8)
+	binary.LittleEndian.PutUint64(eth1DepositIndexBuf, state.eth1DepositIndex)
+	eth1DepositBuf := bytesutil.ToBytes32(eth1DepositIndexBuf)
+	fieldRoots[fieldRootIx] = eth1DepositBuf[:]
+	fieldRootIx++
+
+	// Validators slice root.
+	validatorsRoot, err := stateutil.ValidatorRegistryRoot(state.validators)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not compute validator registry merkleization")
+	}
+	fieldRoots[fieldRootIx] = validatorsRoot[:]
+	fieldRootIx++
+
+	// Balances slice root.
+	balancesRoot, err := stateutil.Uint64ListRootWithRegistryLimit(state.balances)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not compute validator balances merkleization")
+	}
+	fieldRoots[fieldRootIx] = balancesRoot[:]
+	fieldRootIx++
+
+	// RandaoMixes array root.
+	mixes := make([][]byte, len(state.randaoMixes))
+	for i := range mixes {
+		mixes[i] = state.randaoMixes[i][:]
+	}
+	randaoRootsRoot, err := stateutil.ArraysRoot(mixes, fieldparams.RandaoMixesLength)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not compute randao roots merkleization")
+	}
+	fieldRoots[fieldRootIx] = randaoRootsRoot[:]
+	fieldRootIx++
+
+	// Slashings array root.
+	slashingsRootsRoot, err := ssz.SlashingsRoot(state.slashings)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not compute slashings merkleization")
+	}
+	fieldRoots[fieldRootIx] = slashingsRootsRoot[:]
+	fieldRootIx++
+
+	if state.version == version.Phase0 {
+		// PreviousEpochAttestations slice root.
+		prevAttsRoot, err := stateutil.EpochAttestationsRoot(state.previousEpochAttestations)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not compute previous epoch attestations merkleization")
+		}
+		fieldRoots[fieldRootIx] = prevAttsRoot[:]
+		fieldRootIx++
+
+		// CurrentEpochAttestations slice root.
+		currAttsRoot, err := stateutil.EpochAttestationsRoot(state.currentEpochAttestations)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not compute current epoch attestations merkleization")
+		}
+		fieldRoots[fieldRootIx] = currAttsRoot[:]
+		fieldRootIx++
+	}
+
+	if state.version == version.Altair || state.version == version.Bellatrix {
```

Comment:
Im concerned these kind of conditionals are very easy to get wrong...perhaps we can create concise helper functions such as `isPhase0` and then you can do `!isPhase0`

Reply:
this particular usage would be equivalent to `state.version != version.Phase0` which I find easier to read. Even `state.version > version.Phase0` would be better. I think helpers for this could be a source of harder to diagnose bugs down the line